test/controllers/SelectPenaltyController.test.ts

```diff
@@ -40,13 +40,49 @@ describe('SelectPenaltyController', () => {
             } as Appeal,
             navigation: { permissions: [SELECT_THE_PENALTY_PAGE_URI] }
         };
+
+        const app = createApp(applicationData);
+
+        await request(app)
+            .get(SELECT_THE_PENALTY_PAGE_URI)
+            .expect(res => {
+                expect(res.text).to.include('type="radio"');
+                expect(res.text).to.include('value="A0000001"');
+                expect(res.text).to.include('value="A0000002"');
+            });
+    });
+
+    it('GET: should redirect to Company Authentication when user is not authenticated', async () => {
```

Comment:
Didn't we agree to have separate test file which makes GET request to set of paths and expects to get that same redirect for every request made?

Reply:
We did, but then when I attempted to implement this I realised that each controller we'd like to apply this iterative test approach to can have differing binding configurations when calling `createApp`. For example, compare [this line](https://github.com/companieshouse/lfp-appeals-frontend/pull/249/files#diff-a2e1a9f6667fe3d3541902b5f0766ff3R44) from Select Penalty with [this line](https://github.com/companieshouse/lfp-appeals-frontend/pull/249/files#diff-f810a152defbda778868a9b40bcf50beR87) from Review Penalty.

Reply:
Does it really matter what data is in the session since company auth middleware is triggered before view rendering?

I would have thought that only thing middleware is concerned is:
- code session to be populated
- correct navigation entry to be present
- different company number to be in two places

Reply:
It's more about the binding configuration than the session. But I understand what you mean - see updated test.